joint_trajectory_controller/include/joint_trajectory_controller/joint_trajectory_controller.hpp

```diff
@@ -68,15 +66,25 @@ class JointTrajectoryController : public controller_interface::ControllerInterfa
   JointTrajectoryController();
 
   JOINT_TRAJECTORY_CONTROLLER_PUBLIC
-  JointTrajectoryController(
-    const std::vector<std::string> & joint_names,
-    const std::vector<std::string> & write_op_names);
+  explicit JointTrajectoryController(const std::vector<std::string> & joint_names);
 
   JOINT_TRAJECTORY_CONTROLLER_PUBLIC
   controller_interface::return_type
-  init(
-    std::weak_ptr<hardware_interface::RobotHardware> robot_hardware,
-    const std::string & controller_name) override;
+  init(const std::string & controller_name) override;
```

Comment:
It seems that you need this constructor only for tests. What if you use a `friend` class as done for `JointStateController`?

Reply:
You are right, I removed it, and actually I could just add a helper method in `TestableJointTrajectoryController` to set that vector.